docs/tutorials/two_screens.rst

```diff
@@ -0,0 +1,493 @@
+         40.38, 18.76, 10.80,  6.31,  5.02,  0.21,
+    ] * u.dimensionless_unscaled
+
+    rng = np.random.default_rng(seed=12345)
+    phase1 = rng.random(len(brightness1)) * 2.*np.pi
+
+    magnification1 = brightness1 / brightness1.max() * np.exp(1j*phase1)
+
+
+Constructing the main screen
+----------------------------
+
+Create the :py:class:`~screens.screen.Screen1D` object for the main screen.
+
+.. jupyter-execute::
+
+    normal1 = CylindricalRepresentation(1., 90.*u.deg - xi1, 0.).to_cartesian()
+
+    screen1 = Screen1D(normal=normal1,
+                       p=pos1,
+                       v=v_lens1,
+                       magnification=magnification1)
+
+
+The second screen
+====================
+
+Set the second screen's properties.
+
+.. jupyter-execute::
+
+    d_s2 = 0.415 * u.kpc
+    xi2 =  46.1 * u.deg
+    v_lens2 = -3.3 * u.km / u.s
+
+For the second screen, we manually set the position and magnification of the
+single image. In principle, these can be calculated from the coordinates and
+brightness of the millisecond feature in the secondary spectrum.
+
+.. jupyter-execute::
+
+    pos2 = [9.1652957] * u.au
+    magnification2 = 0.1
+
+Create the :py:class:`~screens.screen.Screen1D` object for the second screen.
+
+.. jupyter-execute::
+
+    normal2 = CylindricalRepresentation(1., 90.*u.deg - xi2, 0.).to_cartesian()
+
+    screen2 = Screen1D(normal=normal2,
+                       p=pos2,
+                       v=v_lens2,
+                       magnification=magnification2)
+
+
+Generating observations
+=======================
+
+Use the :py:meth:`~screens.screen.Screen.observe` method to generate two sets
+of optical paths: one for radiation scattered only by the main screen
+(resulting in the main parabola) and one for radiation scattered by both
+screens (yielding the millisecond feature).
+
+.. jupyter-execute::
+
+    obs1 = telescope.observe(
+        source=screen1.observe(source=pulsar, distance=d_p-d_s1),
+        distance=d_s1)
+
+    obs2 = telescope.observe(
+        source=screen1.observe(
+            source=screen2.observe(source=pulsar, distance=d_p-d_s2),
+            distance=d_s2-d_s1),
+        distance=d_s1)
+
+The :py:class:`~screens.screen.Screen1D` class assumes that the linear features
+that cause the images on the lens continue indefinitely. The lens that causes
+the millisecond feature, however, is found to be limited in extent, not
+producing any scatterings beyond a certain point on the sky. To model this, we
+have to use a little hack: we can select optical paths in ``obs2`` based on
+their positions at the main screen. For these positions to be available, they
+first need to be computed. This is triggered by the first line, which computes
+the geometric delays of the optical paths, for which the positions of the
+scattering points on both screens need to be calculated.
+
+.. jupyter-execute::
+
+    obs2.tau
+    bool_on_lens2 = obs2.source.pos.x.squeeze() < 7. * u.au
```

Comment:
I'd use `.ravel()` instead of `.squeeze()` here to make it the same as below. I think the `.squeeze()` actually only works coincidentally, because the pulsar, telescope and screen 2 only have 1 element.

Reply:
Corrected